corehq/apps/case_search/utils.py

```diff
@@ -65,45 +62,31 @@ class CaseSearchProfiler:
     queries: list = field(default_factory=list)
     _query_number: int = 0
 
-    def run_query(self, slug, es_query):
-        self._query_number += 1
-        if self.debug_mode:
-            es_query = es_query.enable_profiling()
-
-        tc = self.timing_context(f'run query #{self._query_number}: {slug}')
-        timer = tc.peek()
-        with tc:
-            results = es_query.run()
-
-        if self.debug_mode:
-            self.queries.append({
-                'slug': slug,
-                'query_number': self._query_number,
-                'query': es_query.raw_query,
-                'duration': timer.duration,
-                'profile_url': self._get_profile_url(slug, self._query_number, results.raw.get('profile')),
-            })
-        return results
+    def get_case_search_class(self, slug=None):
+        profiler = self
+
+        class ProfiledCaseSearchES(CaseSearchES):
+            def run(self):
+                profiler._query_number += 1
+                if profiler.debug_mode:
+                    self.es_query['profile'] = True
+
+                tc = profiler.timing_context(f'run query #{profiler._query_number}: {slug}')
+                timer = tc.peek()
+                with tc:
+                    results = super().run()
 
-    def add_query(self, slug, es_query):
```

Comment:
Was this function just unused?

Reply:
No, it was used to grab profiling context from queries when running it (search for `profiler.add_query` in the diff here), but this PR subclasses `CaseSearchES` to do that a little less obtrusively